mpi4jax/_src/collective_ops/allreduce.py

```diff
@@ -230,3 +278,4 @@ def mpi_allreduce_transpose_rule(tan_args, *x_args, op, comm, transpose):
 # assign to the primitive the correct encoder
 mlir.register_lowering(mpi_allreduce_p, mpi_allreduce_xla_encode_cpu, platform="cpu")
 mlir.register_lowering(mpi_allreduce_p, mpi_allreduce_xla_encode_gpu, platform="cuda")
+mlir.register_lowering(mpi_allreduce_p, mpi_allreduce_xla_encode_xpu, platform="xpu")
```

Comment:
```suggestion
mlir.register_lowering(mpi_allreduce_p, translation_rule_gpu(mpi_allreduce_xla_encode_gpu), platform="cuda")
mlir.register_lowering(mpi_allreduce_p, translation_rule_xpu(mpi_allreduce_xla_encode_gpu), platform="xpu")
```

This should work, right? Since the descriptors are the same structs anyway, and the kernels have the same name?

Reply:
I am working on it and I will propose something because there are slight changes

Reply:
The files `mpi_xla_bridge_gpu.pyx` and `mpi_xla_bridge_xpu.pyx` aren't exactly identical. The differences are quite small. I don't know how to reduce duplication in there.
I tried to get close to your suggestion by writing it like this:
```python
mlir.register_lowering(
    mpi_allreduce_p,
    translation_rule_xpu(
        partial(
            mpi_allreduce_xla_encode_device,
            build_allreduce_descriptor=xla_bridge.mpi_xla_bridge_xpu.build_allreduce_descriptor,
        )
    ),
    platform="xpu",
)
``` 
but it doesn't allow me to use module mpi_xla_bridge_gpu because it doesn't exist when installing for xpu only.

Reply:
>  build_allreduce_descriptor=xla_bridge.mpi_xla_bridge_xpu.build_allreduce_descriptor

I don't get that one, isn't the XPU descriptor exactly the same as the GPU descriptor?